xbmc/platform/darwin/ios/network/NetworkIOS.mm

```diff
@@ -0,0 +1,496 @@
+{
+   struct ifreq ifr;
+   strcpy(ifr.ifr_name, m_interfaceName.c_str());
+   if (ioctl(m_network->GetSocket(), SIOCGIFFLAGS, &ifr) < 0)
+      return false;
+
+   return ((ifr.ifr_flags & IFF_UP) == IFF_UP);
+}
+
+bool CNetworkInterfaceIOS::IsConnected() const
+{
+   struct ifreq ifr;
+   int zero = 0;
+   memset(&ifr,0,sizeof(struct ifreq));
+   strcpy(ifr.ifr_name, m_interfaceName.c_str());
+   if (ioctl(m_network->GetSocket(), SIOCGIFFLAGS, &ifr) < 0)
+      return false;
+
+   // ignore loopback
+   int iRunning = ( (ifr.ifr_flags & IFF_RUNNING) && (!(ifr.ifr_flags & IFF_LOOPBACK)));
+
+   if (ioctl(m_network->GetSocket(), SIOCGIFADDR, &ifr) < 0)
+      return false;
+
+   // return only interfaces which has ip address
+   return iRunning && (0 != memcmp(ifr.ifr_addr.sa_data+sizeof(short), &zero, sizeof(int)));
+}
+
+std::string CNetworkInterfaceIOS::GetMacAddress() const
+{
+  return m_interfaceMacAdr;
+}
+
+void CNetworkInterfaceIOS::GetMacAddressRaw(char rawMac[6]) const
+{
+  memcpy(rawMac, m_interfaceMacAddrRaw, 6);
+}
+
+std::string CNetworkInterfaceIOS::GetCurrentIPAddress(void) const
+{
+   std::string result;
+
+   struct ifreq ifr;
+   strcpy(ifr.ifr_name, m_interfaceName.c_str());
+   ifr.ifr_addr.sa_family = AF_INET;
+   if (ioctl(m_network->GetSocket(), SIOCGIFADDR, &ifr) >= 0)
+   {
+      result = inet_ntoa((*((struct sockaddr_in *)&ifr.ifr_addr)).sin_addr);
+   }
+  
+  if (StringUtils::StartsWith(m_interfaceName, "pdp_ip"))
+  {
+    result += " (cellular)";
+  }
+  else if (StringUtils::StartsWith(m_interfaceName, "utun"))
+  {
+    result += " (vpn)";
+  }
+  else if (StringUtils::StartsWith(m_interfaceName, "en"))
+  {
+    result += " (wifi)";
+  }
+
+   return result;
+}
+
+std::string CNetworkInterfaceIOS::GetCurrentNetmask(void) const
+{
+   std::string result;
+
+   struct ifreq ifr;
+   strcpy(ifr.ifr_name, m_interfaceName.c_str());
+   ifr.ifr_addr.sa_family = AF_INET;
+   if (ioctl(m_network->GetSocket(), SIOCGIFNETMASK, &ifr) >= 0)
+   {
+      result = inet_ntoa((*((struct sockaddr_in*)&ifr.ifr_addr)).sin_addr);
+   }
+
+   return result;
+}
+
+std::string CNetworkInterfaceIOS::GetCurrentWirelessEssId(void) const
+{
+   std::string result;
+   return result;
+}
+
+const struct sockaddr *GetSockAddr(const struct rt_msghdr2 *rtm)
+{
+  //sockaddrs are after the message header
+  const struct sockaddr* dst_sa = reinterpret_cast<const struct sockaddr *>(rtm + 1);
+  
+  if((rtm->rtm_addrs & (RTA_DST | RTA_GATEWAY)) == (RTA_DST | RTA_GATEWAY))
+  {
+    if(dst_sa[RTAX_DST].sa_family == AF_INET &&
+       !((rtm->rtm_flags & RTF_WASCLONED) &&
+         (rtm->rtm_parentflags & RTF_PRCLONING)))
+    {
+      
+      if (dst_sa[RTAX_GATEWAY].sa_family == AF_INET)
+      {
+        return &dst_sa[RTAX_GATEWAY];
+      }
+    }
+  }
+  return nullptr;
+}
+
+bool GetGatewayForIfName(std::string ifName, struct sockaddr **foundSockAddr)
+{
+  bool result = false;
+  size_t needed = 0;
+  int mib[6];
+  char *buf = nullptr;
```

Comment:
https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md#52-pointer-and-reference-types
```suggestion
  char* buf = nullptr;
```